### PR TITLE
added isErrorMessage function

### PR DESCRIPTION
**Pull request by kostola**
While parsing messages from a ExchangeFeed, I had to filter for errors and I didn't find a dedicated function. So I added it.

### CODE REVIEW

**Comment by kostola** [comment]
@Ike-Clinton thanks! Do you want me to include a test for this change in order to merge it or are you referring only to future changes?

**Comment by Ike-Clinton** [comment]
Not up to me! :)

**Comment by kostola** [comment]
@Ike-Clinton sorry. I can't assign the PR but I saw that @CjS77 or @mihar merged previous PRs. I hope you can find the time to take a look at this one. Thanks ;)

File: src/core/Messages.ts [hunk 1]
```diff
@@ -33,6 +33,10 @@ export interface ErrorMessage {
     message: string;
 }
 
+export function isErrorMessage(msg: any): boolean {
+    return !!msg.message && !!msg.message.message && typeof msg.message.message === 'string';
```
Comment (CjS77) [inline src/core/Messages.ts]: Basically, you should only have `!!msg.message` here (and maybe the typecheck, but the typescript compiler is supposed to make such checks less critical, so I'm inclined to leave it out too). Reason being that by this stage, all messages should be standardised, and must thus conform to the `ErrorMessage` interface above.

If you're seeing cases where this isn't happening, it should be flagged as a bug.

**Comment by CjS77** [comment]
@kostola, where are you filtering for errors? Once messages are in the stream, or from the underlying WS API?

I don't think we use `ErrorMessage` anywhere yet, so there's an opportunity to redefine it at this point to capture any use cases that fit.

For one thing, I'm thinking of changing the interface definition to
```
export interface ErrorMessage extends StreamMessage {
    message: string;
    details?: any
}
```
Then it fits into the streaming message system more elegantly and also allows users to pass metadata on (something that will possible expand to other messages).

If you're filtering from the API level, then the message should be coerced into the defined interface; or an 'error' message should be emitted (not to be confused with an error message that is sent along the message stream) and handled appropriately by its instantiator.

**Comment by kostola** [comment]
@CjS77 I modified the method as you suggested.
I am filtering messages coming from an ExchangeFeed, in particular authentication error messages.
I left the type check because also `UnknownMessage` interface contains a `message` field of type `any`.

**Comment by CjS77** [comment]
Perfect sense. So what we really need to do is have those messages handled and pushed as `ErrorMessage` instances rather than embedded in `UnknownMessage`. 

I'll merge your PR in now, but I'll open an issue to implement the above-mentioned change, but will let you carry on in the meantime.

**Pull request merged**